lib/atomic_cache/storage/store.rb

```diff
@@ -26,6 +26,18 @@ def set(key, new_value, user_options); raise NotImplementedError end
       # returns true if it succeeds; false otherwise
       def delete(key, user_options); raise NotImplementedError end
 
+      protected
+
+      def marshal(value, user_options={})
```

Comment:
curious, but would we not assume that the underlying cache impl is marshaling for us?

https://github.com/petergoldstein/dalli/blob/master/lib/dalli/client.rb#L29
https://github.com/petergoldstein/dalli/blob/master/lib/dalli/server.rb#L408

Reply:
That is exactly what I assumed when I first wrote this, which is why it lacked marshaling the first time around.  When Lucas and I started on the first integration, it was clear that the previous `Rails.cache.fetch` was doing serialization, and AtomicCache was not.  Granted, we were utilizing the in-memory storage adapter, so at a minimum that likely needs to support it.

Since I knew it needed to be added for in-memory, I poked around the Rails source to see if they were handling it for their version of the memcache client, which is when I turned up this:  https://github.com/rails/rails/blob/master/activesupport/lib/active_support/cache/mem_cache_store.rb#L194

I had implemented marshaling for both the in-memory and the dalli storage adapter, primarily because I was inferring it was needed based on the Rails source.

So, "curious" indeed.  Based on the Dalli source you've linked to, I'd be fine pulling out marshaling for the Dalli storage adapter, but leaving it in for the in-memory adapter.